flake.nix

```diff
@@ -9,24 +9,23 @@
         nixpkgs = self;
       };
 
-      lib = import ./lib;
+      libVersionInfoOverlay = import ./lib/flake-version-info.nix self;
+      lib = (import ./lib).extend libVersionInfoOverlay;
 
       forAllSystems = lib.genAttrs lib.systems.flakeExposed;
     in
     {
+      overlays.setLibVersionInfo = final: prev: {
+        lib = prev.lib.extend libVersionInfoOverlay;
+      };
```

Comment:
```suggestion
```

This is not a Nixpkgs overlay, but a `lib` overlay. It seems unnecessary, and confusing.

We can discuss it more, or first merge without this, because it is not essential.

Reply:
Fine for me, done.